lvmd/command/lvm.go

```diff
@@ -223,49 +207,56 @@ func (g *VolumeGroup) CreateVolume(ctx context.Context, name string, size uint64
 		}
 	}
 	lvcreateArgs = append(lvcreateArgs, lvcreateOptions...)
-	lvcreateArgs = append(lvcreateArgs, g.Name())
+	lvcreateArgs = append(lvcreateArgs, vg.Name())
+
+	return callLVM(ctx, lvcreateArgs...)
+}
 
-	if err := callLVM(ctx, "lvcreate", lvcreateArgs...); err != nil {
+// FindPool finds a named thin pool in this volume group.
+func (vg *VolumeGroup) FindPool(ctx context.Context, name string) (*ThinPool, error) {
+	pools, err := vg.ListPools(ctx, name)
+	if err != nil {
 		return nil, err
 	}
-	if err := g.Update(ctx); err != nil {
```

Comment:
The reason Update never needs to be called is because the state that LVM is in after the change is never actually used by lv/vgservice. There is no use in updating a state internally if it is never actually used by the code. Sure, canonically it may have made sense to update it, but effectively this was dead code since the updated values were never useful (except for certain update scenarios like resizing which are really specific)

Reply:
Thank you for your explanation. Now I understand this better.